stack-graphs/src/cycles.rs

```diff
@@ -129,71 +129,150 @@ where
 // ----------------------------------------------------------------------------
 // Cycle detector
 
-pub trait Appendable {
-    type Ctx;
+pub struct Appendables<H> {
+    /// List arena for appendable lists
+    elements: ListArena<InternedPathOrHandle<H>>,
+    /// Arena for interned partial paths
+    interned: Arena<PartialPath>,
+}
+
+impl<H> Appendables<H> {
+    pub fn new() -> Self {
+        Self {
+            elements: ListArena::new(),
+            interned: Arena::new(),
+        }
+    }
+}
 
-    fn append_to(
+#[derive(Clone)]
+pub(crate) enum InternedPathOrHandle<H> {
```

Comment:
If it’s not already taken up in this namespace, I feel like `Interned` would be a better, or at least more concise, name for this, since it’s already got a handle type parameter.

Reply:
This name was a bit awkward. I've renamed it to `InternedOrHandle` (the grouping was `(InternedPath)OrHandle`, which was clearly ambiguous).

Note that this type is strictly internal to the `cycles` module (it wasn't, but could be, so I've changed that), so not part of the public API.